woocommerce/woocommerce
# Prevent adding global_unique_id to old schema versions

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This adds a guard to ensure the checkout continues to run successfully in case the database update fails for some reason, during WooCommerce update.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

I couldn't think of a way of testing this without database access, as it's a very specific scenario to reproduce.

1. Go to `WooCommerce > Status > Tools > Verify base database tables` and click on 'Verify database'
2. Execute a checkout for a product successfully
3. Using WP cli, run the command `wp option update woocommerce_schema_version 430` to update the option to 430.
4. Using a database tool, delete the `global_unique_id` column from the `wp_wc_product_meta_lookup` table
5. Execute a checkout for a product successfully again

At the end of testing, make sure to run `Verify database` again.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
